datascan/gcloud-commands/spec-files/data-quality-spec.yaml

```diff
@@ -0,0 +1,125 @@
+data_quality_spec:
+  # [Optional] The percentage of the records to be selected from the dataset for DataScan.
+  # If present, this variable will override the variable provided in datascan_defaults.
+  sampling_percent: 100
+
+  # [Optional]  A filter applied to all rows in a single DataScan job. The filter needs to be a valid SQL expression for a WHERE clause in BigQuery standard SQL syntax. Example: col1 >= 0 AND col2 < 10
+  # If present, this variable will override the variable provided in datascan_defaults.
+  row_filter: "station_id > 1000"
+
+  # [Optional]  The configuration of BigQuery export post scan action upon job completion.
+  post_scan_actions:
+    bigquery_export:
+      results_table: "//bigquery.googleapis.com/projects/bigquery-public-data/datasets/austin_bikeshare/tables/export_table"
+
+  # DataQualityScan rule configurations. Documentation for the supported rule types and rule specifications can be found in https://cloud.example.com/dataplex/docs/reference/rest/v1/DataQualityRule.
+  # Can optionally reference templated rules provided in the variable datascan_rules_library_file_path. Any variables provided here will override the default values provided in the rules library file.
+  rules:
+    - column: address
+      dimension: VALIDITY
+      ignore_null: null
+      non_null_expectation: {}
+      threshold: 0.99
+    - column: council_district
+      dimension: VALIDITY
+      ignore_null: true
+      threshold: 0.9
+      range_expectation:
+        max_value: '10'
+        min_value: '1'
+        strict_max_enabled: false
+        strict_min_enabled: true
+    - column: council_district
+      dimension: VALIDITY
+      range_expectation:
+        max_value: '9'
+        min_value: '3'
+      threshold: 0.8
+    - column: power_type
+      dimension: VALIDITY
+      ignore_null: false
+      regex_expectation:
+        regex: .*solar.*
+    - column: property_type
+      dimension: VALIDITY
+      ignore_null: false
+      set_expectation:
+        values:
+        - sidewalk
+        - parkland
+    - column: address
+      dimension: UNIQUENESS
+      uniqueness_expectation: {}
+    - column: number_of_docks
+      dimension: VALIDITY
+      statistic_range_expectation:
+        max_value: '15'
+        min_value: '5'
+        statistic: MEAN
+        strict_max_enabled: true
+        strict_min_enabled: true
+    - column: footprint_length
+      dimension: VALIDITY
+      row_condition_expectation:
+        sql_expression: footprint_length > 0 AND footprint_length <= 10
+    - dimension: VALIDITY
+      table_condition_expectation:
+        sql_expression: COUNT(*) > 0
+
+# [Optional] Prefix for DataScan name. 
+# If present, this variable will override the variable provided in datascan_defaults.
+# prefix: "test"
+
+# [Optional] Custom description for DataScan.
+# If present, this variable will override the variable provided in datascan_defaults.
```

Comment:
A lot of these fields are terraform specific (example -- IAM bindings); 
Some of them have documentation comments which are TF specific (example -- labels).

Let's use it as a starting point, but make sure that the fields and comments make sense for the gcloud YAML spec.

Reply:
To think about it, spec only supports rules and specs like row filter, sampling etc. So removed other fields